test/testdata/java_shelf_name.baseline

```diff
@@ -114,28 +114,28 @@ public class ShelfName implements ResourceName {
 
   @Override
   public String toString() {
-    return PATH_TEMPLATE.instantiate("shelf_id", shelfId);
+    return PATH_TEMPLATE.instantiate("shelf", shelf);
   }
 
   /** Builder for ShelfName. */
   public static class Builder {
 
-    private String shelfId;
+    private String shelf;
 
-    public String getShelfId() {
-      return shelfId;
+    public String getShelf() {
```

Comment:
Are we forced to change the `{shelf_id}` path substring into `{shelf}`? It changes this public method name.

Reply:
Yes - if we don't, it changes the inferred class name (e.g. you get ShelfIdName). This change is actually what we want to be consistent with most (all??) APIs in googleapis.

Reply:
Ok, this is a breaking change for the GA Firestore client. While i think the impact is very small, I'm not sure how we should handle versioning that client.

Reply:
Maybe we can work around it with entity_name overrides?